base/abstractarray.jl

```diff
@@ -258,7 +249,20 @@ julia> strides(A)
 (1, 3, 12)
 ```
 """
-strides(A::AbstractArray) = size_to_strides(1, size(A)...)
+function strides end
+
+# the definition of strides for Array is the cumsum of the product of sizes
```

Comment:
cumprod of the size?  But it's not quite that, it's `cumprod([1;collect(size(A))[1:end-1]])`